ic-stable-structures/src/structure/cache/multimap.rs

```diff
@@ -0,0 +1,196 @@
+use std::cell::RefCell;
+use std::hash::Hash;
+
+use ic_exports::stable_structures::{memory_manager::MemoryId, BoundedStorable};
+use mini_moka::unsync::{Cache, CacheBuilder};
+
+use crate::structure::*;
+
+/// A LRU Cache for StableMultimaps
+pub struct CachedStableMultimap<K1, K2, V>
+where
+    K1: BoundedStorable + Clone + Hash + Eq + PartialEq + Ord,
+    K2: BoundedStorable + Clone + Hash + Eq + PartialEq + Ord,
+    V: BoundedStorable + Clone,
+{
+    inner: StableMultimap<K1, K2, V>,
+    cache: RefCell<Cache<(K1, K2), V>>,
+}
+
+impl<K1, K2, V> CachedStableMultimap<K1, K2, V>
+where
+    K1: BoundedStorable + Clone + Hash + Eq + PartialEq + Ord,
+    K2: BoundedStorable + Clone + Hash + Eq + PartialEq + Ord,
+    V: BoundedStorable + Clone,
+{
+    /// Create new instance of the CachedStableMultimap with a fixed number of max cached elements.
+    pub fn new(memory_id: MemoryId, max_cache_items: u64) -> Self {
+        Self::with_map(StableMultimap::new(memory_id), max_cache_items)
+    }
+
+    /// Create new instance of the CachedStableMultimap with a fixed number of max cached elements.
+    pub fn with_map(inner: StableMultimap<K1, K2, V>, max_cache_items: u64) -> Self {
+        Self {
+            inner,
+            cache: RefCell::new(
+                CacheBuilder::default()
+                    .max_capacity(max_cache_items)
+                    .build(),
+            ),
+        }
+    }
+}
+
+impl<K1, K2, V> MultimapStructure<K1, K2, V> for CachedStableMultimap<K1, K2, V>
+where
+    K1: BoundedStorable + Clone + Hash + Eq + PartialEq + Ord,
+    K2: BoundedStorable + Clone + Hash + Eq + PartialEq + Ord,
+    V: BoundedStorable + Clone,
+{
+    fn get(&self, first_key: &K1, second_key: &K2) -> Option<V> {
+        let mut cache = self.cache.borrow_mut();
+        let key = (first_key.clone(), second_key.clone());
+
+        match cache.get(&key) {
```

Comment:
maybe these nested match cases could be simplified with unwrap_or_else functions

Reply:
Done